lib/libcore/cell.rs

```diff
@@ -1672,10 +1675,13 @@ impl<T: ?Sized> UnsafeCell<T> {
     #[inline]
     #[unstable(feature = "unsafe_cell_raw_get", issue = "66358")]
     pub const fn raw_get(this: *const Self) -> *mut T {
-        // We can just cast the pointer from `UnsafeCell<T>` to `T` because of
-        // #[repr(transparent)]. This exploits libstd's special status, there is
-        // no guarantee for user code that this will work in future versions of the compiler!
-        this as *const T as *mut T
+        const fn crucible_hook<T: ?Sized>(this: *const UnsafeCell<T>) -> *mut T {
+            // We can just cast the pointer from `UnsafeCell<T>` to `T` because of
+            // #[repr(transparent)]. This exploits libstd's special status, there is
+            // no guarantee for user code that this will work in future versions of the compiler!
```

Comment:
Do you have any thoughts for a future-proof version of this?

Reply:
That comment is copied from the original libstd sources, so it's not really ours to future-proof.  I don't think we'll have much trouble, though - we have a custom override for this function, which should keep working even if rustc and/or libstd change the implementation.

Reply:
Oh, okay! The fact that it was in `crucible_hook` made me think it was custom for Crux. That all sounds good.

Reply:
The `crucible_hook` is just to give us an unambiguous name to override.  This module has several functions named `get`, so an override for `core::cell::{{impl}}::get` would match them all.  (The full names actually contain a disambiguating index on the `{{impl}}` part, but those indices change frequently, so we don't want to hard-code them.)  Instead, we wrap the body of the one function we care about in a uniquely-named inner function, and override that instead.  There is only one `core::cell::{{impl}}::get::crucible_hook`, so this is unambiguous.  (If we wanted to override several different `get` methods, we'd have to give each hook a different name.  We used to do this in `core::slice`, but it looks like those overrides are all gone now, thanks to the unsafe code improvements.)

Reply:
Oh, right, that makes sense.

Reply:
I'll add a comment to this effect - I think the "hook" mechanism was only documented in those now-removed slice overrides.

Reply:
👍